cpg_utils/hail_batch.py

```diff
@@ -226,3 +257,56 @@ def authenticate_cloud_credentials_in_job(
 
     # activate the google service account
     job.command(GCLOUD_AUTH_COMMAND)
+
+
+def query_command(
+    module,
+    func_name: str,
+    *func_args,
+    setup_gcp: bool = False,
+    hail_billing_project: Optional[str] = None,
+    hail_bucket: Optional[str] = None,
+    default_reference: str = 'GRCh38',
+    packages: Optional[List[str]] = None,
+) -> str:
+    """
+    Run a Python Hail Query function inside a Hail Batch job.
+    Constructs a command string to use with job.command().
+    If hail_billing_project is provided, Hail Query will be initialised.
+    """
+    python_cmd = f"""
+import logging
+logger = logging.getLogger(__file__)
+logging.basicConfig(format='%(levelname)s (%(name)s %(lineno)s): %(message)s')
+logger.setLevel(logging.INFO)
+"""
+    if hail_billing_project:
+        assert hail_bucket
+        python_cmd += f"""
+import asyncio
+import hail as hl
+asyncio.get_event_loop().run_until_complete(
+    hl.init_batch(
+        default_reference='{default_reference}',
+        billing_project='{hail_billing_project}',
+        remote_tmpdir='{hail_bucket}',
```

Comment:
When initiating a hail Batch with a remote tmpdir, does Hail check that the path exists? If so, this CMD would need to come after the GCloud Auth command. I'll have a play locally

Reply:
the remote_tmpdir is just stored using String manipulation, no problem